src/main/java/com/google/devtools/build/lib/actions/ArtifactExpander.java

```diff
@@ -24,12 +24,27 @@
 /** Expands tree artifacts and filesets. */
 public interface ArtifactExpander {
 
+  /**
+   * Returns the expansion of the given {@linkplain SpecialArtifactType#TREE tree artifact}.
+   *
+   * @throws MissingExpansionException if this expander does not have data for the given tree
+   *     artifact
+   */
+  ImmutableSortedSet<TreeFileArtifact> expandTreeArtifact(Artifact treeArtifact)
+      throws MissingExpansionException;
+
   /**
    * Returns the expansion of the given {@linkplain SpecialArtifactType#TREE tree artifact}.
    *
    * <p>If this expander does not have data for the given tree artifact, returns an empty set.
    */
-  ImmutableSortedSet<TreeFileArtifact> expandTreeArtifact(Artifact treeArtifact);
+  default ImmutableSortedSet<TreeFileArtifact> expandTreeArtifactOrEmpty(Artifact treeArtifact) {
```

Comment:
`tryExpandTreeArtifact`?

Reply:
I applied the change, but wonder whether that coul be understood as "try to expand, but if that fails return the tree artifact *unchanged*"? That could be misleading in a subtle way.

Reply:
Not perfect but I think it's ok given the return type and documentation.

Reply:
I do wonder whether we could switch all the callers to the strict version, or at least push the try/catch into the callers that really need it. Not saying that you should attempt that - this PR is already an improvement over the current state.

Reply:
We can do that as a follow-up, but that would seem like a change that a Blaze developer should make. If it causes any breakages, they probably won't be caught by Bazel presubmit.